flake.nix

```
@@ -96,8 +96,8 @@
             program =
               let
                 ubuntu = pkgs.fetchurl {
-                  url = "https://cloud-images.ubuntu.com/jammy/20231027/jammy-server-cloudimg-amd64.img";
-                  hash = "sha256-a7Ukf4eRm4A8IRr9GvdLMJa+boNNrCnPrHEdrXLq/qg=";
+                  url = "http://cloud-images.ubuntu.com/releases/jammy/release-20231211/ubuntu-22.04-server-cloudimg-amd64.img";
```

Looks like the TLS endpoint also works here. Do you believe the `http` version is more stable?

It probably doesn't matter much since we verify integrity with the hash.

```suggestion
                  url = "https://cloud-images.ubuntu.com/releases/jammy/release-20231211/ubuntu-22.04-server-cloudimg-amd64.img";
```

---

Oh, I didn't think about this at all. Good catch!